extensions/wikia/AdEngine/js/lookup/prebid/adaptersPerformanceTracker.js

```diff
@@ -0,0 +1,88 @@
+define('ext.wikia.adEngine.lookup.prebid.adaptersPerformanceTracker', [
+	'ext.wikia.adEngine.adTracker',
+	'wikia.window'
+], function (adTracker, win) {
+	'use strict';
+
+	var emptyResponseMsg = 'EMPTY',
+		notRespondedMsg = 'NOT_RESPONDED';
+
+	function setupPerformanceMap(skin, adapters) {
+		var biddersPerformanceMap = {};
+
+		adapters.forEach(function (adapter) {
+			var slots = adapter.getSlots(skin),
+				adapterName = adapter.getName();
+
+			if (adapter.isEnabled()) {
+				Object.keys(slots).forEach(function (slotName) {
+					biddersPerformanceMap[adapterName] = biddersPerformanceMap[adapterName] || {};
+					biddersPerformanceMap[adapterName][slotName] = notRespondedMsg;
+				});
+			}
+		});
+
+		return biddersPerformanceMap;
+	}
+
+	function updatePerformanceMap(performanceMap) {
+		var allBids;
+
+		if (typeof win.pbjs.getBidResponses === 'function') {
+			allBids = win.pbjs.getBidResponses();
+
+			Object.keys(allBids).forEach(function (slotName) {
+				var slotBids = allBids[slotName].bids;
+
+				slotBids.forEach(function (bid) {
+					performanceMap[bid.bidder][slotName] = getParamsFromBidForTracking(bid);
+				});
+			});
+		}
+
+		return performanceMap;
+	}
+
+	function trackBidderSlotState(adapter, slotName, providerName, performanceMap) {
+		var bidderName = adapter.getName(), category;
+
+		if (adapter.isEnabled()) {
```

Comment:
Don't you think early returns would make the code more readable rather than if-else tree?